lib/iris/experimental/ugrid/__init__.py

```diff
@@ -17,6 +17,7 @@
 from functools import wraps
 import re
 import threading
+import warnings
```

Comment:
For the ugrid module, @bjlittle and I have been so far been deliberately avoiding `warnings.warn()` - **in favour of `logger.debug()`** - on the understanding that this pattern is being gradually propagated throughout Iris to make the user experience less 'noisy'.

Reply:
Well, I considered that.  But I was trying to be consistent with other things in the netcdf loading code -- in the Pyke arena, mostly.

Reply:
I see. Should I therefore raise a separate pull request to convert the existing code in this module to use `warnings.warn()` instead of `logger.debug()` ([e.g. L3336](https://github.com/SciTools/iris/blob/6e329d28fe40cf07735d95e7ad3730851216570d/lib/iris/experimental/ugrid/__init__.py#L3336))?

Reply:
Well, I'd expect it to be more likely the other way about : we should eventually converge on logging for everything.
I was just hoping to be consistent with other parts of the loading code, for now.
But now I realise that I don't know enough about what has been done elsewhere recently.

I also see you are also using logger.debug rather than logger.warning, which I don't really get -- aren't these warning-level problems?  Does this represent a new intention to be silent unless the user specifically asks about it ?
I was expecting to see a message by default, but make it easy to hide.

Reply:
>Does this represent a new intention to be silent unless the user specifically asks about it ?

Yeah that's where this is coming from. To be fair to you I was also reticent about this when talking to Bill, but it's what has already been adopted during the cube arithmetic work.

This deserves a team conversation soon if examples have already gone into `master` (cube arithmetic), but we're not all on the same page.

I don't know what to do here. Happy to leave it to @pp-mo and @stephenworsley to judge.

Reply:
Here's my plan : 
  * use logging in place of warnings
  * but use logging.warning instead of logging.debug/info in this case.
     * looking at other usages, I think it is perfectly consistent to use different levels for different cases (though I do still think that some of the existing levels are wrong)
  * it also means developing a means of testing for a logged message [in place of a warning](https://github.com/SciTools/iris/blob/6e329d28fe40cf07735d95e7ad3730851216570d/lib/iris/tests/integration/experimental/test_ugrid_load.py#L160).  
     So, I'll put this PR into draft + re-enable it when I've fixed that ...

Reply:
>though I do still think that some of the existing levels are wrong

Agreed

> it also means developing a means of testing for a logged message [in place of a warning](https://github.com/SciTools/iris/blob/6e329d28fe40cf07735d95e7ad3730851216570d/lib/iris/tests/integration/experimental/test_ugrid_load.py#L160).

There are plenty of examples testing for logged `DEBUG` messages in the existing experimental.ugrid unit tests 👍

Reply:
> examples testing for logged DEBUG messages

oh, thanks for the hint @trexfeathers !